src/Apps/Collect/CollectionApp.tsx

```diff
@@ -13,8 +14,15 @@ interface CollectionAppProps {
 }
 
 export class CollectionApp extends Component<CollectionAppProps> {
+  collectionNotFound = collection => {
+    if (!collection) {
+      throw new HttpError(404)
+    }
+  }
+
   render() {
     const { collection } = this.props
+    this.collectionNotFound(collection)
```

Comment:
I would move that to `componentDidMount` or `componentWillReceiveProps`, basically move it out of render